### PR TITLE
[MRG] Implement MondrianForestRegressor

### PR DESCRIPTION
**Pull request by MechCoder**

### CODE REVIEW

**Comment by MechCoder** [comment]
@balajiln Do you keep boostrap on or off in the MondrianForest implementation?

**Comment by balajiln** [comment]
I switch off bootstrap.

**Comment by MechCoder** [comment]
Updated to MRG. Should be ready for a first pass.

File: mondrian_forest/ensemble/forest.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+        The prediction is a GMM drawn from
+        \sum_{i=1}^T w_i N(m_i, \sigma_i) where w_i = 1 / T.
+
+        The mean E[Y | X] reduces to 1 / T * \sum_{i=1}^T m_i
+
+        The variance Var[Y | X] is given by
+            = E[Y**2 | X] - E[Y | X]**2
+            = (1 / T) * \sum_{i=1}^T E[Y**2_i| X] - E[Y| X]**2
+            = (1 / T) * \sum_{i=1}^T (Var[Y_i | X] + E[Y_i | X]**2) - E[Y| X]**2
```
Comment (glouppe) [inline mondrian_forest/ensemble/forest.py]: @balajiln Is this what you had in mind? (That looks good to me given the description in your thesis, but I prefer to be sure)

File: mondrian_forest/ensemble/tests/test_forest.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+boston = load_boston()
+scaler = MinMaxScaler()
+X, y = boston.data, boston.target
+X = scaler.fit_transform(X)
+
+# The time of split and feature chosen for splitting are highly
+# scale-sensitive.
+scaler = MinMaxScaler()
+X = scaler.fit_transform(X)
+
```
Comment (glouppe) [inline mondrian_forest/ensemble/tests/test_forest.py]: Could you add tests checking the correctness of the mean and std? I think this is the most important thing here.

Comment (glouppe) [inline mondrian_forest/ensemble/tests/test_forest.py]: (Also checking the corner cases, e.g. those that lead to std==0, test data outside of training data, etc)

**Comment by MechCoder** [comment]
Added those tests. I will try reproducing the benchmarks in a while but doing that should be independent of merging this.

File: mondrian_forest/ensemble/forest.py (hunk 1 above)
Comment (balajiln) [inline mondrian_forest/ensemble/forest.py]: Yes, this is correct.

**Comment by glouppe** [comment]
+1 for merge
